cargo-miri/src/phases.rs

```diff
@@ -284,6 +284,11 @@ pub fn phase_rustc(mut args: impl Iterator<Item = String>, phase: RustcPhase) {
         // For Windows, do the same thing again with `.exe` appended to the filename.
         // (Need to do this here as cargo moves that "binary" to a different place before running it.)
         info.store(&out_filename("", ".exe"));
+
+        // cargo-miri never runs in wasm, so the --target should always be set when compiling for wasm.
+        if non_host_target_crate().map_or(false, |target| target.contains("wasm")) {
```

Comment:
Let's keep this consistent... both here and below in the `emit_link_hack` section we currently just emit all the extensions; if we want to make that target-dependent we should do that for all extensions.